public/app/plugins/datasource/loki/components/LokiLabel.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { forwardRef, HTMLAttributes } from 'react';
+import { cx, css } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme } from '@grafana/ui';
+// @ts-ignore
+import Highlighter from 'react-highlight-words';
+
+/**
+ * @public
+ */
+export type OnLabelClick = (name: string, value: string | undefined, event: React.MouseEvent<HTMLElement>) => any;
+
+export interface Props extends Omit<HTMLAttributes<HTMLElement>, 'onClick'> {
+  name: string;
+  active?: boolean;
+  loading?: boolean;
+  searchTerm?: RegExp;
+  value?: string;
+  facets?: number;
+  onClick?: OnLabelClick;
+}
+
+export const LokiLabel = forwardRef<HTMLElement, Props>(
+  ({ name, value, hidden, facets, onClick, className, loading, searchTerm, active, style, ...rest }, ref) => {
+    const theme = useTheme();
+    const styles = getLabelStyles(theme, name);
+
+    const onLabelClick = (event: React.MouseEvent<HTMLElement>) => {
+      if (onClick && !hidden) {
+        onClick(name, value, event);
+      }
+    };
+    let text = value || name;
+    if (facets) {
+      text = `${text} (${facets})`;
+    }
+
+    return (
+      <span
+        key={text}
+        ref={ref}
+        onClick={onLabelClick}
+        style={style}
+        title={text}
+        role="option"
+        aria-selected={!!active}
+        className={cx(
+          styles.base,
+          active && styles.active,
+          loading && styles.loading,
+          hidden && styles.hidden,
+          className,
+          onClick && !hidden && styles.hover
+        )}
+        {...rest}
+      >
+        <Highlighter textToHighlight={text} searchWords={[searchTerm]} highlightClassName={styles.matchHighLight} />
+      </span>
+    );
+  }
+);
+
+const getLabelStyles = (theme: GrafanaTheme, name: string) => {
+  const colors = ['#FF7368', '#459EE7'];
```

Comment:
I see that these colors are used as a border colors for the tags. I am wondering if we can't use a red & blue color from our palette/colors because this doesn't seem consistent. Or we can use `getTagStyles` and style this as tags.